### PR TITLE
resolve: Refactor away record_exports, fixing #4953

### PR DESCRIPTION
**Pull request by jseyfried**

### CODE REVIEW

File: src/librustc_resolve/resolve_imports.rs [hunk 1]
```diff
@@ -938,6 +949,18 @@ impl<'a, 'b:'a, 'tcx:'b> ImportResolver<'a, 'b, 'tcx> {
...
+        let node_id = match module.def_id() {
+            Some(def_id) => self.resolver.ast_map.as_local_node_id(def_id).unwrap(),
+            None => return,
+        };
+        let export = match binding.def() {
+            Some(def) => Export { name: name, def_id: def.def_id() },
+            None => return,
+        };
+        self.resolver.export_map.entry(node_id).or_insert(Vec::new()).push(export);
+    }
```
Comment (nrc) [inline src/librustc_resolve/resolve_imports.rs]: Could you move the if public check into add_export

**Comment by nrc** [comment]
Wow, nice.

r+ with the tiny bit of refactoring

File: src/librustc_resolve/resolve_imports.rs (hunk 1 above)
Comment (jseyfried) [inline src/librustc_resolve/resolve_imports.rs]: Sure

**Comment by jseyfried** [comment]
@nrc Thanks for the feedback. I moved the public check into add_export and amended the commit.

**Comment by jseyfried** [comment]
@nrc Rebased

**Pull request merged**